tests/types/test_schema.py

```diff
@@ -987,3 +997,180 @@ def test_infer_param_schema_with_errors():
+                        ]
+                    )
+                ),
+            )
+        ]
+    )
+
+    data = [
+        {
+            "messages": [
+                {"role": "system", "content": "Translate every message you receive to French."},
+                {"role": "user", "content": "I like machine learning"},
+            ],
+        },
+        {
+            "messages": [
+                {"role": "user", "content": "Summarize the following document ...", "name": "jeff"},
+            ]
+        },
+    ]
+    dtype = _infer_colspec_type(data)
+    assert dtype == Array(
+        Object(
+            [
+                Property(
+                    "messages",
+                    Array(
+                        Object(
+                            [
+                                Property("role", DataType.string),
+                                Property("content", DataType.string),
+                                Property("name", DataType.string, required=False),
+                            ]
+                        )
+                    ),
+                )
+            ]
+        )
+    )
+    schema = Schema([ColSpec(dtype)])
+    assert schema.to_dict() == [
+        {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": [
+                    {
+                        "name": "messages",
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": [
+                                {"name": "content", "type": "string", "required": True},
+                                {"name": "name", "type": "string", "required": False},
+                                {"name": "role", "type": "string", "required": True},
+                            ],
+                        },
+                        "required": True,
+                    }
+                ],
+            },
+            "required": True,
+        }
+    ]
+    # assert schema.to_dict() == [
+    #     {
+    #         "type": "array",
+    #         "items": {
+    #             "type": "object",
+    #             "properties": {
+    #                 "messages": {
+    #                     "type": "array",
+    #                     "items": {
+    #                         "type": "object",
+    #                         "properties": {
+    #                             "content": {"type": "string", "required": True},
+    #                             "name": {"type": "string", "required": False},
+    #                             "role": {"type": "string", "required": True},
+    #                         },
+    #                     },
+    #                     "required": True,
+    #                 }
+    #             },
+    #         },
+    #         "required": True,
+    #     }
+    # ]
```

Comment:
What is the behavior for invalid json being passed? Will it be very clear (with an MLflow message explaining to the user what they did wrong) for this case?

Reply:
Good point! I'll add test cases covering them